src/components/pages/details/Details.tsx

```diff
@@ -1,32 +1,201 @@
-        setData(response.data);
-      } catch (error) {
-        console.error('Error fetching data:', error);
+    const fetchDetails = async () => {
+      if (plantID) {
+        const details = await fetchBotanicPlantDetails(plantID);
+        setPlantDetails(details.data);
+        setLoading(false);
       }
-    }
+    };
+
+    fetchDetails();
+  }, [plantID]);
+
+  const renderPlantDetail = ({ label, value, Icon }: PlantDetailProps) => (
+    <Typography
+      variant="body1"
+      sx={{
+        display: 'flex',
+        alignItems: 'center',
+        fontWeight: '500',
+        fontSize: { xs: '0.9rem', md: '1.25rem' },
+        textAlign: { xs: 'center', md: 'left' },
+        lineHeight: '116%',
+        marginBottom: '1.25rem',
+      }}
+    >
+      <Icon />
+      {`${label}: ${value}`}
+    </Typography>
+  );
+
+  const renderPlantDetails = () => {
+    if (!plantDetails) return null;
 
-    fetchData();
-  }, [API_KEY]);
+    return (
+      <Grid container spacing={3}>
+        <Grid
+          item
+          xs={12}
+          md={6}
+          sx={{
+            display: 'flex',
+            flexDirection: 'column',
+            justifyContent: 'center',
+            alignItems: { xs: 'center', md: 'flex-start' },
+            textAlign: { xs: 'center', md: 'left' },
+            paddingLeft: { xs: '0.625rem', md: '3.125rem', mb: '5rem' },
+          }}
+        >
+          <Typography
+            component="h2"
+            variant="h3"
+            sx={{
+              fontWeight: '500',
+              fontSize: { xs: '2rem', md: '3rem' },
+              lineHeight: '116%',
+              marginBottom: '2.5rem',
+            }}
+          >
+            {plantDetails.common_name.charAt(0).toUpperCase() +
+              plantDetails.common_name.slice(1)}
+          </Typography>
+          {renderPlantDetail({
+            label: 'Sunlight',
+            value: plantDetails.sunlight,
+            Icon: Icons.WbSunny,
+          })}
+          {renderPlantDetail({
+            label: 'Plant Familly',
+            value: plantDetails.family,
+            Icon: Icons.Diversity3,
+          })}
+          {renderPlantDetail({
+            label: 'Watering',
+            value: plantDetails.watering,
+            Icon: Icons.WaterDrop,
+          })}
+          {renderPlantDetail({
+            label: 'Origin',
+            value: plantDetails.origin,
+            Icon: Icons.Language,
+          })}
+        </Grid>
+        <Grid item xs={12} md={6}>
+          <CardMedia
+            sx={{
+              borderRadius: '3.75rem',
+              width: '100%',
+              height: 'auto',
+              maxWidth: '100%',
+            }}
+            component="img"
+            src={plantDetails.default_image.medium_url}
+            alt="Plant"
+          />
+        </Grid>
+      </Grid>
+    );
+  };
 
-  console.log(data);
   return (
-    <div>
-      <h2>Details about plant</h2>
-    </div>
+    <main>
```

Comment:
we already have main tag (https://github.com/VitiukVV/plant-care-quide/blob/main/src/components/appBar/AppBar.tsx#L107) :)